diff -up ./CMakeLists.txt.pre-rmo ./CMakeLists.txt
--- ./CMakeLists.txt.pre-rmo	2016-05-15 20:15:36.000000000 -0700
+++ ./CMakeLists.txt	2016-05-15 19:59:41.000000000 -0700
@@ -0,0 +1,81 @@
+PROJECT(acrosync)
+
+INCLUDE(FindOpenSSL)
+
+INCLUDE_DIRECTORIES( 
+    ${OPENSSL_INCLUDE_DIR}
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${srcsync_BINARY_DIR}/usr/local/include
+    ${srcsync_BINARY_DIR}/usr/local/include/libssh
+)
+
+LINK_DIRECTORIES( 
+    ${OPENSSL_LIB_DIR}
+    ${srcsync_BINARY_DIR}/usr/local/lib /usr/local/opt/libiconv/lib
+)
+
+ADD_LIBRARY(acrosync SHARED
+
+    rsync/rsync_client.cpp
+    rsync/rsync_entry.cpp
+    rsync/rsync_file.cpp
+    rsync/rsync_io.cpp
+    rsync/rsync_log.cpp
+    rsync/rsync_pathutil.cpp
+    rsync/rsync_socketio.cpp
+    rsync/rsync_socketutil.cpp
+    rsync/rsync_sshio.cpp
+    rsync/rsync_stream.cpp
+    rsync/rsync_timeutil.cpp
+    rsync/rsync_util.cpp
+)
+
+TARGET_LINK_LIBRARIES( acrosync 
+    ${OPENSSL_LIBRARIES}
+    ssh2
+    iconv
+)
+
+ADD_EXECUTABLE(test-acrosync 
+    rsync/t_rsync_client.cpp
+)
+
+TARGET_LINK_LIBRARIES( test-acrosync 
+
+    acrosync
+    iconv
+)
+
+
+
+INSTALL(
+    FILES
+        rsync/rsync_client.h
+        rsync/rsync_entry.h
+        rsync/rsync_file.h
+        rsync/rsync_io.h
+        rsync/rsync_log.h
+        rsync/rsync_pathutil.h
+        rsync/rsync_socketio.h
+        rsync/rsync_socketutil.h
+        rsync/rsync_sshio.h
+        rsync/rsync_stream.h
+        rsync/rsync_timeutil.h
+        rsync/rsync_util.h
+
+    DESTINATION include/rsync/
+)
+
+INSTALL(
+    FILES
+        block/block_out.h
+
+    DESTINATION include/block/
+)
+
+INSTALL(
+    TARGETS
+        acrosync
+
+    LIBRARY DESTINATION lib
+)
